.github/workflows/wheels.yml

```diff
@@ -0,0 +1,30 @@
+name: Build wheels
+
+on: [push, pull_request]
+
+jobs:
+  build_wheels:
+    name: Build wheels on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-20.04, macOS-10.15]
+
+    steps:
+      - uses: actions/checkout@v2
+
+      # Used to host cibuildwheel
+      - uses: actions/setup-python@v2
+
+      - name: Install cibuildwheel
+        run: python -m pip install cibuildwheel==2.3.0
+
+      - name: Build wheels
+        run: python -m cibuildwheel --output-dir wheelhouse
+        # to supply options, put them in 'env', like:
+        # env:
+        #   CIBW_SOME_OPTION: value
+
+      - uses: actions/upload-artifact@v2
+        with:
+          path: ./wheelhouse/*.whl
```

Comment:
Having a source distribution that runs `make configure` might be nice. Something as

    build_sdist:
      name: Build source distribution
      runs-on: ubuntu-latest
      steps:
      - uses: actions/checkout@v2
      - uses: actions/setup-python@v2
        name: Install Python
      - name: make configure
        run: make configure
      - name: Build sdist
        run: python setup.py sdist
      - uses: actions/upload-artifact@v2
        with:
          path: dist/*.tar.gz

I don't know if this exactly works, but it might prevent future pushes to pypi to avoid this mistake. I use it for `memory_allocator`:

https://github.com/sagemath/memory_allocator/blob/main/.github/workflows/build_wheels.yml

Reply:
the wheel itself is built in a docker container. While it's possible to customise what's happening there, as I did in 
https://github.com/dimpase/primecountpy/blob/db814b23ce1b92c0347dba23d64505a9a75d63dd/.github/workflows/wheels.yml#L25 (to line 42), I don't see what it buys us here.

Reply:
The idea would be that we have all assets from one workflow file, like here:

https://github.com/sagemath/memory_allocator/releases

The tarball that github provides doesn't work for installation in sage (and maybe not elsewhere), because the sage installation assumes that `configure` is contained in the tarball. That is why `cysignals` is currently on version 1.11.1. on pypi.

Reply:
the tarball in question is the PyPI tarball. The builder

1) spins a docker container for each (`c*-manylinux-*`, etc) platform
2) pip-installs, in place `cysignals` there
3) builds the wheel
4) runs `auditwheel` on it to make sure it's correctly packaging libraries etc
5) stores the result in "artefacts"

You can install cibuildwheel locally, in docker, to get a better idea of what it does, if you like:
https://cibuildwheel.readthedocs.io/en/stable/setup/#local

Reply:
It seems we have a misunderstanding. Your pull request is mostly fine.

In addition to have wheels-distributions, I think it makes sense to distribute the sources as well. When distributing the source (in a `.tar.gz`), one needs to include the configure file obtained with `make configure`. Otherwise the `.tar.gz` won't work for sage to install the package.

So I'm suggesting adding another job. However, you can just reject this for now.

Reply:
In another PR. Besides, it's something that can easily and quickly be done locally, so it's not that urgent.